brainrender_napari/widgets/atlas_manager_view.py

```diff
@@ -0,0 +1,118 @@
+        self.hideColumn(
+            self.model().column_headers.index("Raw name")
+        )  # hide raw name
+
+    def _on_row_double_clicked(self):
+        atlas_name = self.selected_atlas_name()
+        if atlas_name in get_downloaded_atlases():
+            up_to_date = get_atlases_lastversions()[atlas_name]["updated"]
+            if not up_to_date:
+                update_dialog = AtlasManagerDialog(atlas_name, "Update")
+                update_dialog.ok_button.clicked.connect(
+                    self._on_update_atlas_confirmed
+                )
+                update_dialog.exec()
+        else:
+            download_dialog = AtlasManagerDialog(atlas_name, "Download")
+            download_dialog.ok_button.clicked.connect(
+                self._on_download_atlas_confirmed
+            )
+            download_dialog.exec()
+
+    def _on_download_atlas_confirmed(self):
+        """Downloads the currently selected atlas and signals this."""
+        atlas_name = self.selected_atlas_name()
+        worker = self._apply_in_thread(install_atlas, atlas_name)
+        worker.returned.connect(self.download_atlas_confirmed.emit)
+        worker.start()
+
+    def _on_update_atlas_confirmed(self):
+        """Updates the currently selected atlas and signals this."""
+        atlas_name = self.selected_atlas_name()
+        worker = self._apply_in_thread(update_atlas, atlas_name)
+        worker.returned.connect(self.update_atlas_confirmed.emit)
+        worker.start()
+
+    def selected_atlas_name(self) -> str:
+        """A single place to get a valid selected atlas name."""
+        selected_index = self.selectionModel().currentIndex()
+        assert selected_index.isValid()
+        selected_atlas_name_index = selected_index.siblingAtColumn(0)
+        selected_atlas_name = self.model().data(selected_atlas_name_index)
+        return selected_atlas_name
+
+    @thread_worker
+    def _apply_in_thread(self, apply: Callable, atlas_name: str):
+        """Calls `apply` on the given atlas in a separate thread."""
+        apply(atlas_name)
+        self.model().refresh_data()
+        return atlas_name
+
+    @classmethod
+    def get_tooltip_text(cls, atlas_name: str):
+        """Returns the atlas name as a formatted string,
+        as well as instructions on how to interact with the atlas."""
+        if atlas_name in get_downloaded_atlases():
+            is_up_to_date = get_atlases_lastversions()[atlas_name]["updated"]
+            if is_up_to_date:
+                tooltip_text = f"{format_atlas_name(atlas_name)} is up-to-date"
+            else:  # needs updating
+                tooltip_text = (
+                    f"{format_atlas_name(atlas_name)} (double-click to update)"
+                )
+        elif atlas_name in get_all_atlases_lastversions().keys():
+            tooltip_text = (
+                f"{format_atlas_name(atlas_name)} (double-click to download)"
+            )
```

Comment:
Is it worth caching the results of `get_downloaded_atlases()`, `get_atlases_lastversions()`, and `get_all_atlases_lastversions()` somewhere and updating them when `model().referesh_data()` is called?

Same for https://github.com/brainglobe/brainrender-napari/blob/37d93e89cf6a72d3847f4cde32e0ac9a0675c8ad/brainrender_napari/widgets/atlas_viewer_view.py#L108-L118

Reply:
Good idea - opened #127 to consider this.